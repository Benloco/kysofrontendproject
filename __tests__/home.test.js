/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render, screen,cleanup,fireEvent } from '@testing-library/react'
import Home from '../pages/index'
import renderer from 'react-test-renderer';

afterEach(cleanup);

jest.mock('next/router', () => ({
    __esModule: true,
    useRouter: jest.fn()
  }))
  
const useRouter = jest.spyOn(require('next/router'), 'useRouter')

jest.mock('next/router', () => require('next-router-mock'));

describe('Home page',()=>{
  it("renders Home component without crashing",()=>{
    render(<Home/>)
  })

  it("should find welcome message",()=>{
    const {container}= render(<Home/>)
      const h2Tag = container.querySelector('h2');
      expect(h2Tag).toHaveTextContent('Kyso Frontend Interview Project');
  })

  it("should display my name",()=>{
      const {getByRole} = render(<Home/>);
      expect(getByRole("heading",{level:6})).toHaveTextContent('Ben Quarshie');
  })

  it("renders button correctly",()=>{
     const {getByRole}= render(<Home/>);
     const element =getByRole("button");
     expect(element).toBeInTheDocument();
     expect(element).toHaveTextContent('View Reports');
    //fireEvent.click(element);
  })

  it("should navigate accordingly",async () => {
    const { getByTestId,getByRole }=  render(<Home />);
    
    const button = getByRole("button");
    fireEvent.click(button)
    useRouter.mockImplementationOnce(() => ({
        push:"/dashboard",
      }))
   

      await expect(useRouter.push);
    
  })


  it("matches snapshot",()=>{
     const homeTree= renderer.create(<Home/>).toJSON();
     expect(homeTree).toMatchSnapshot();
  })
})