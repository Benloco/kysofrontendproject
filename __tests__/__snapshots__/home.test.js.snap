// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<div
  className="mt-6 col-lg-6 col-md-6 col-sm-12 col-md offset-lg-3 offset-md-2 indexContent"
>
  <h2>
    Kyso Frontend Interview Project
  </h2>
  <div>
    <h6
      className="mt-5"
    >
      by Ben Quarshie
    </h6>
    <button
      className="mt-5 btn btn-primary btn-lg"
      onClick={[Function]}
    >
      View Reports
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 "
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;
